crates/matrix-sdk-crypto/src/olm/group_sessions/sender_data.rs

```diff
@@ -74,6 +74,9 @@ pub enum SenderData {
         /// If false, we had simply accepted the key as this user's latest
         /// key.
         master_key_verified: bool,
+
+        #[serde(default)]
+        identity_needs_user_approval: bool,
```

Comment:
I added this so that session_data_finder can find this information, so that the trust level can be calculated.  But I think that it would be better to combine this with the `master_key_verified` field, and make them into a new `master_key_status` field that is an enum with `Unverified`, `NeedsApproval`, and `Verified` options.  Maybe we should even add another option for the case where the user's previous identity was verified, but they rotated their identity.  Maybe something like `PreviousIdentityVerified`.

What do you think?

(If you think it's better to keep it as-is, then I'll write documentation for this field, but I didn't want to write it if I was going to remove it right away.)

Reply:
You've asked me for my opinion on this question, but this change doesn't seem to appear anywhere in the PR, so I don't think I have enough context on the change to form a sensible opinion.